pymongo/common.py

```diff
@@ -611,17 +684,41 @@ def validate(option, value):
     return lower, value
 
 
-def get_validated_options(options, warn=True):
-    """Validate each entry in options and raise a warning if it is not valid.
-    Returns a copy of options with invalid entries removed
+def get_validated_options(options):
```

Comment:
Technically all these public methods and globals are public so we need to consider backwards compatibility. Why remove the "warn" option?